indexer/indexing.py

```diff
@@ -92,66 +97,98 @@ async def create_pg_vector_index_if_not_exists(self):
+                    UploadFile(
+                        file=io.BytesIO(file_content),
+                        size=os.path.getsize(file_path),
+                        filename=os.path.basename(file_path),
+                    )
                 )
-                counter += 1
+            else:
+                content = await self.text_converter.textify(file_path)
+                for chunk in self.splitter.split_text(content):
+                    new_metadata = {
+                        "chunk-id": str(counter),
+                        "document_name": file,
+                    }
+                    source_chunks.append(
+                        Document(page_content=chunk, metadata=new_metadata)
+                    )
+                    counter += 1
         try:
-            if os.environ["OPENAI_API_TYPE"] == "azure":
-                embeddings = AzureOpenAIEmbeddings(
-                    model="text-embedding-ada-002",
-                    azure_deployment=os.environ["AZURE_DEPLOYMENT_NAME"],
-                    azure_endpoint=os.environ["AZURE_OPENAI_ENDPOINT"],
-                    openai_api_type=os.environ["OPENAI_API_TYPE"],
-                    openai_api_key=os.environ["AZURE_OPENAI_API_KEY"],
-                )
+            if os.getenv("INDEXER_TYPE") == "r2r":
+                r2r_app = self.get_r2r()
+                await r2r_app.engine.aingest_files(files=source_files)
             else:
-                embeddings = OpenAIEmbeddings(client="")
-            db = PGVector.from_documents(
-                embedding=embeddings,
-                documents=source_chunks,
-                collection_name=indexer_input.collection_name,
-                connection=self.db_url,
-                pre_delete_collection=True,  # delete collection if it already exists
-            )
+                embeddings = self.get_embeddings()
+                db = PGVector.from_documents(
+                    embedding=embeddings,
+                    documents=source_chunks,
+                    collection_name=indexer_input.collection_name,
+                    connection_string=self.db_url,
+                    pre_delete_collection=True,
+                )
+                self.collection_name = db.collection_name
+                await self.create_pg_vector_index_if_not_exists()
             print(
-                f"Embeddings have been created for the collection: {db.collection_name}"
+                f"Embeddings have been created for the collection: {self.collection_name}"
             )
-            await self.create_pg_vector_index_if_not_exists()
         except Exception as e:
             raise InternalServerException(e.__str__())
 
+    def get_r2r(self):
+        if os.getenv("OPENAI_API_TYPE") == "azure":
+            embedding_provider = AzureOpenAIEmbeddingProvider(
+                EmbeddingConfig(
+                    provider="azure-openai",
+                    base_model="text-embedding-3-large",
```

Comment:
These model names have to be taken in as parameters (from Queue Message) and cannot be hardcoded. It can be ENV variables but not hardcoded.

Reply:
Yes sure.